### PR TITLE
[Rebased] Add file type support and update core to 7.0.2 SDK-4096

### PR DESCRIPTION
**Pull request by CTLalit**
- [x] Add `CleverTap.defineFileVariable("fileVariable")`
- [x] Add `CleverTap.onOneTimeVariablesChanged(handler)`
- [x] Add `CleverTap.onVariablesChangedAndNoDownloadsPending(handler)`
- [x] Add `CleverTap.onceVariablesChangedAndNoDownloadsPending(handler)`
- [x] Add `CleverTap.onFileValueChanged('fileVariable', handler)`
- [x] Support core SDK `7.0.2`
- [x] Update [Variables.md](https://github.com/CleverTap/clevertap-react-native/blob/master/docs/Variables.md)
- [x] Update [ChangeLog.md](https://github.com/CleverTap/clevertap-react-native/blob/master/CHANGELOG.md)
- [x] Update Example App
- [x] Bump lib version to `3.1.0`

### CODE REVIEW

**Review by darshanclevertap — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,27 @@
...
+  * Supports [CleverTap iOS SDK v7.0.2](https://github.com/CleverTap/clevertap-ios-sdk/blob/master/CHANGELOG.md#version-702-october-10-2024).
+  * Adds support for custom handshake domains.
+
+* **[Android and iOS Platform]**
+  * Adds support for File Type Variables in Remote Config. Please refer to the [Remote Config Variables](https://github.com/CleverTap/clevertap-ios-sdk/blob/master/docs/Variables.md) doc to read more on how to integrate this in your app.
+  * Adds support for Custom Code Templates. Please refer to the [CustomCodeTemplates.md](https://github.com/CleverTap/clevertap-android-sdk/blob/master/docs/CustomCodeTemplates.md) doc to read more on how to integrate this in your app.
+  * Adds support for custom code in-app templates definitions through a json scheme.
+
+**Bug Fixes**
+* **[Android and iOS Platform]**
```
Comment (Anush-Shand) [inline CHANGELOG.md:22]: Let's also mention these github issues in the changelog
https://github.com/CleverTap/clevertap-react-native/issues/431
https://github.com/CleverTap/clevertap-react-native/issues/426

**Review by nzagorchev — CHANGES_REQUESTED** [review]
Please address the changes requested in #430 and add fixes to the current branch (`feat/file-vars-rebased`).

File: CHANGELOG.md (hunk 1 above)
Comment (piyush-kukadiya) [reply CHANGELOG.md:22]: Fixed

**Review by piyush-kukadiya — APPROVED** [review]
LGTM

**Review by nzagorchev — APPROVED** [review]

**Review by vasct — APPROVED** [review]

**Review by Anush-Shand — APPROVED** [review]

**Pull request merged**